extensions/positron-r/src/provider.ts

```diff
@@ -25,123 +25,101 @@ export const R_DOCUMENT_SELECTORS = [
+	updateBinaries(userHqBinaries);
 
-	const pathBinary = await findRBinaryFromPATH();
-	if (pathBinary && !binaries.has(pathBinary)) {
-		binaries.set(pathBinary, BinarySource.PATH);
-	}
+	const userMoreBinaries = discoverAdHocBinaries(userRBinaries());
+	updateBinaries(userMoreBinaries);
 
-	// make sure we include the "current" version of R, for some definition of "current"
-	// we've probably already discovered it, but we still want to single it out, so that we mark
-	// that particular R installation as the current one
-	const curBin = await findCurrentRBinary();
-	if (curBin) {
-		rInstallations.push(new RInstallation(curBin, true));
-		binaries.delete(curBin);
+	// (Try to) promote each RBinary to a proper RInstallation
+	let rInstallations: Array<RInstallation> = [];
+	if (currentBinaries.length > 0) {
+		const currentBinary = currentBinaries[0];
+		rInstallations.push(new RInstallation(currentBinary.path, true, currentBinary.reasons));
+		binaries.delete(currentBinary.path);
 	}
 
-	binaries.forEach((source, bin) => {
-		rInstallations.push(new RInstallation(bin));
+	binaries.forEach(rbin => {
+		rInstallations.push(new RInstallation(rbin.path, false, rbin.reasons));
 	});
 
-	// TODO: possible location to tell the user why certain R installations are being omitted from
-	// the interpreter drop-down and, in some cases, offer to help fix the situation:
-	// * version < minimum R version supported by positron-r
-	// * (macOS only) version is not orthogonal and is not the current version of R
-	// * invalid R installation
+	const rejectedRInstallations: Array<RInstallation> = [];
 	rInstallations = rInstallations
 		.filter(r => {
-			if (!r.valid) {
-				LOGGER.info(`Filtering out ${r.binpath}: invalid R installation.`);
-				return false;
-			}
-			return true;
-		})
-		.filter(r => {
-			if (!(r.current || r.orthogonal)) {
-				LOGGER.info(`Filtering out ${r.binpath}: not current and also not orthogonal.`);
-				return false;
-			}
-			return true;
-		})
-		.filter(r => {
-			if (!r.supported) {
-				LOGGER.info(`Filtering out ${r.binpath}: version is < ${MINIMUM_R_VERSION}`);
+			if (!r.usable) {
+				LOGGER.info(`Filtering out ${r.binpath}, reason: ${friendlyReason(r.reasonRejected)}.`);
+				rejectedRInstallations.push(r);
 				return false;
 			}
 			return true;
 		});
 
-	// FIXME? should I explicitly check that there is <= 1 R installation
-	// marked as 'current'?
+	if (rejectedRInstallations.length > 0) {
+		if (rInstallations.length === 0) {
+			LOGGER.warn(`All discovered R installations are unusable by Positron.`);
+			LOGGER.warn('Learn more about R discovery at https://positron.posit.co/r-installations.html');
+			vscode.window.showWarningMessage(
+				`All discovered R installations are unusable by Positron.`,
+				{ modal: true }, 'See the log file')
+				.then(selection => {
+					if (selection === 'See the log file') {
+						LOGGER.show();
+					}
+				});
```

Comment:
I hope to get feedback on:

* Should this be `model: true` or `false` (the default)? If `true`, the warning cannot be ignored or missed; the user must engage with it. If `false`, it's more subtle and eventually goes away on its own.
* Should we only throw this warning ... once? Once per workspace? Once every week? If someone *wants* to use Positron and R, they see this warning, they fix the situation, and live happily ever after. But if someone has zero interest in R, but also wants to keep some unusable R installations around, they might find this warning annoying.

Reply:
Here's what `model: true` looks like. User must click on "See the log file" or "Cancel" to proceed.

<img width="1387" alt="Screenshot 2024-12-06 at 2 50 05 PM" src="https://github.com/user-attachments/assets/676205b3-ea57-4eff-8d9f-c365d188c4b5">

Reply:
Here's what it looks like without `modal: true`. I've added the orange rectangle just to emphasize the pop-up. It's certainly missable and it eventually goes away on its own.

<img width="1387" alt="Screenshot 2024-12-06 at 2 52 09 PM" src="https://github.com/user-attachments/assets/1d244eb7-0363-490a-a528-a9a534294ce1">

Reply:
I vote for `modal: true`, given that the user is just dead in the water in this situation.